src/amo/components/Link.js

```diff
@@ -0,0 +1,44 @@
+import path from 'path';
+
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { Link } from 'react-router';
+
+
+export class LinkBase extends React.Component {
+  static propTypes = {
+    base: PropTypes.string,
+    children: PropTypes.node,
+    to: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    base: '',
+  }
+
+  render() {
+    const { base, children, to } = this.props;
+
+    let linkTo;
+    if (typeof to === 'string') {
+      linkTo = path.join(base, to);
+    } else if (to && to.pathname) {
+      linkTo = { ...to, pathname: path.join(base, to.pathname) };
+    } else {
+      linkTo = to;
+    }
+
```

Comment:
I'm not too sure about the use-case for supporting the object approach.

It might be worth making sure the string output for LinkTo still starts with base after the join since it would be odd if the base part could be clobbered, for that you'd just supply a different base prop.